app/src/main/java/com/github/se/signify/model/quest/QuestRepositoryFireStore.kt

```diff
@@ -21,7 +21,9 @@ class QuestRepositoryFireStore(private val db: FirebaseFirestore) : QuestReposit
     db.collection(collectionPath).get().addOnCompleteListener { task ->
       if (task.isSuccessful) {
         val quests =
-            task.result?.mapNotNull { document -> documentToQuest(document) } ?: emptyList()
+            task.result
```

Comment:
get the document of the firestore and sorted by indexes to get a valid order. Good job